### PR TITLE
added support for RFC5806

### PR DESCRIPTION
**Pull request by ValerieMoreaureau**

### CODE REVIEW

File: sip-ri/src/main/java/gov/nist/javax/sip/header/ParameterNames.java [hunk 1]
```diff
@@ -167,4 +167,12 @@ public interface ParameterNames {
     public static final String LOCAL_TAG = "local-tag";
     public static final String REMOTE_TAG = "remote-tag";
 
+    //Support for RFC 5806 @MukhaValeriia
+	public static final String REASON = "reason";
+	public static final String LIMIT = "limit";
+	public static final String COUNTER = "counter";
+	public static final String PRIVACY = "privacy";
+	public static final String SCREEN = "screen";
+	public static final String EXTENSION = "extension";
```
Comment (deruelle) [inline sip-ri/src/main/java/gov/nist/javax/sip/header/ParameterNames.java:176]: This one doesn't exist in the spec, it's a token so a regular param, see the syntax at https://datatracker.ietf.org/doc/html/rfc5806#section-4

diversion-extension = token ["=" (token | quoted-string)]

File: sip-ri/src/main/java/gov/nist/javax/sip/header/extensions/Diversion.java [hunk 2]
```diff
@@ -0,0 +1,363 @@
...
+    *
+    * @param reason diversion-limit parameter value to set
+    * @throws ParseException if the provided parameter is invalid
+    */
+   public void setLimit(String l) throws ParseException {
+   	  if (l == null)
+             throw new NullPointerException("no limit param");
+         else if (l.trim().equals(""))
+             throw new ParseException("bad parameter", 0);
+         this.setParameter(ParameterNames.LIMIT, l);
```
Comment (deruelle) [inline sip-ri/src/main/java/gov/nist/javax/sip/header/extensions/Diversion.java:119]: We can add a check to ensure it match the syntax of 2DIGIT at https://datatracker.ietf.org/doc/html/rfc5806#section-4 i.e: diversion-limit = "limit" "=" 1*2DIGIT

File: sip-ri/src/main/java/gov/nist/javax/sip/header/extensions/Diversion.java [hunk 3]
```diff
@@ -0,0 +1,363 @@
...
+    * Sets the diversion-counter parameter of the Diversion header.
+    *
+    * @param reason diversion-counter parameter value to set
+    * @throws ParseException if the provided parameter is invalid
+    */
+   public void setCounter(String c) throws ParseException {
+   	  if (c == null)
+             throw new NullPointerException("no counter param");
+         else if (c.trim().equals(""))
+             throw new ParseException("bad parameter", 0);
```
Comment (deruelle) [inline sip-ri/src/main/java/gov/nist/javax/sip/header/extensions/Diversion.java:156]: Maybe the parameter of that method shouldn't be a String or we can add a check to ensure it match the syntax of 2DIGIT at https://datatracker.ietf.org/doc/html/rfc5806#section-4 i.e: diversion-counter = "counter" "=" 1*2DIGIT

File: sip-ri/src/main/java/gov/nist/javax/sip/header/extensions/Diversion.java [hunk 4]
```diff
@@ -0,0 +1,363 @@
...
+       return getParameter(ParameterNames.EXTENSION);
+   }
+   
+   /**
+    * Sets the diversion-extension parameter of the Diversion header.
+    *
+    * @param reason diversion-extension parameter value to set
+    * @throws ParseException if the provided parameter is invalid
+    */
+   public void setExtension(String ex) throws ParseException {
```
Comment (deruelle) [inline sip-ri/src/main/java/gov/nist/javax/sip/header/extensions/Diversion.java:267]: I think the extension methods can be removed and use regular setParameter methods for those given the syntax of extensions 
diversion-extension = token ["=" (token | quoted-string)]

File: sip-ri/src/test/java/gov/nist/javax/sip/parser/extensions/DiversionParserTest.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    }
+
+    public void testParseDiversionHeaderParameters() throws ParseException {
+        String diversionHeader = "Diversion: sip:user@example.com;reason=busy;limit=4;privacy=conditionally;counter=2;screen=yes;extension=token\n";
+        try {
+            DiversionParser parser = new DiversionParser(diversionHeader);
+            DiversionList diversionList = (DiversionList) parser.parse();
+            assertNotNull(diversionList);
+
+            for (DiversionHeader diversion : diversionList) {
```
Comment (deruelle) [inline sip-ri/src/test/java/gov/nist/javax/sip/parser/extensions/DiversionParserTest.java:26]: You should not only print them but validate them with assert to ensure they match the value that was passed in the initial diversionHeader String and test the toString()/encode() too to ensure it properly output the same String

**Review by deruelle — CHANGES_REQUESTED** [review]
@ValerieMoreaureau provided my comments. can you please review?

File: sip-ri/src/main/java/gov/nist/javax/sip/header/ParameterNames.java (hunk 1 above)
Comment (ValerieMoreaureau) [reply sip-ri/src/main/java/gov/nist/javax/sip/header/ParameterNames.java:176]: Do we need to check in general the extension of the header? Should I add a check to the test and in parser?

File: sip-ri/src/main/java/gov/nist/javax/sip/header/extensions/Diversion.java (hunk 2 above)
Comment (ValerieMoreaureau) [reply sip-ri/src/main/java/gov/nist/javax/sip/header/extensions/Diversion.java:119]: Done

File: sip-ri/src/main/java/gov/nist/javax/sip/header/extensions/Diversion.java (hunk 4 above)
Comment (ValerieMoreaureau) [reply sip-ri/src/main/java/gov/nist/javax/sip/header/extensions/Diversion.java:267]: done

File: sip-ri/src/main/java/gov/nist/javax/sip/header/ParameterNames.java (hunk 1 above)
Comment (deruelle) [reply sip-ri/src/main/java/gov/nist/javax/sip/header/ParameterNames.java:176]: given diversion-extension = token ["=" (token | quoted-string)] you can check in the tests that additional random parameters provided by the application can be still parsed and included such as ;random-param=random-value by example

**Review by deruelle — CHANGES_REQUESTED** [review]

**Review by deruelle — APPROVED** [review]

**Pull request merged**